daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1459,10 +1459,9 @@ private[lf] object SBuiltin {
         case Right((keyMapping, next)) =>
           onLedger.ptx = onLedger.ptx.copy(contractState = next)
           keyMapping match {
-            case ContractStateMachine.KeyActive(coid)
-                if onLedger.ptx.contractState.locallyCreated.contains(coid) =>
+            case ContractStateMachine.KeyActive(coid) =>
               machine.checkKeyVisibility(onLedger, gkey, coid, operation.handleKeyFound)
-            case _ =>
+            case ContractStateMachine.KeyInactive =>
```

Comment:
This is the crux of the fix

Reply:
I'm confused, isn’t the crux dropping of the `if` in line 1463 not this?

Reply:
You are right.